File: 53. Maximum Subarray/Solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int max = nums[0];
        int sum = 0;
        int start = 0;
        int arr_start,arr_end,i;
        for(i =0 ; i<n ;i++){
            
            if(sum == 0){
                start = i;
            }

            sum = sum + nums[i];

            if(sum > max){
                max = sum;
                arr_start = start;
                arr_end = i;
            } 

            if(sum < 0){
                sum = 0;
            }
        }

        return max;
    }
};

File: 53. Maximum Subarray/Solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        max = nums[0]
        
        if n == 0:
            return []

        sum = 0
        for i in range(n):
            sum = sum + nums[i]

            if sum > max:
                max = sum  
            
            if sum < 0:
                sum = 0
        
        return max


#https://youtu.be/AHZpyENo7k4